File: RailgunVB/Commands/Zap.vb
Imports Discord
Imports Discord.Commands
Imports RailgunVB.Core.Utilities

Namespace Commands
    
    <Group("zap")>
    Public Class Zap
        Inherits ModuleBase
        
        Private ReadOnly _client As IDiscordClient
        Private ReadOnly _commandUtils As CommandUtils

        Public Sub New(client As IDiscordClient, commandUtils As CommandUtils)
            _client = client
            _commandUtils = commandUtils
        End Sub
        
        <Command>
        Public Async Function ZapAsync(Optional user As IGuildUser = Nothing) As Task
            If user IsNot Nothing AndAlso user.Id = _client.CurrentUser.Id
                await ReplyAsync("I'm immune to electricity, BAKA!")
                Return
            End If
            
            Dim name As String = Await _commandUtils.GetUsernameOrMentionAsync(If(user, Context.User))
            await ReplyAsync($"{Format.Bold(name)} has been electrocuted! Something smells nice doesn't it?")
        End Function
        
    End Class
    
End NameSpace

File: Railgun/Commands/Fun/Zap.cs
using System.Threading.Tasks;
using Discord;
using Finite.Commands;
using Railgun.Core.Commands;
using Railgun.Core.Utilities;

namespace Railgun.Commands.Fun
{
    [Alias("zap")]
    public class Zap : SystemBase
    {
        private readonly IDiscordClient _client;
        private readonly CommandUtils _commandUtils;

        public Zap(IDiscordClient client, CommandUtils commandUtils) {
            _client = client;
            _commandUtils = commandUtils;
        }
        
        [Command]
        public async Task ZapAsync(IGuildUser user = null) {
            if (user != null && user.Id == _client.CurrentUser.Id) {
                await ReplyAsync("I'm immune to electricity, BAKA!");
                return;
            }
            
            var name = await _commandUtils.GetUsernameOrMentionAsync(user ?? (IGuildUser)Context.Author);

            await ReplyAsync($"{Format.Bold(name)} has been electrocuted! Something smells nice doesn't it?");
        }
    }
}